llvm/lib/Transforms/InstCombine/InstructionCombining.cpp

```diff
@@ -3661,8 +3661,7 @@ Instruction *InstCombinerImpl::visitUnconditionalBranchInst(BranchInst &BI) {
 
   auto GetLastSinkableStore = [](BasicBlock::iterator BBI) {
     auto IsNoopInstrForStoreMerging = [](BasicBlock::iterator BBI) {
-      return BBI->isDebugOrPseudoInst() ||
-             (isa<BitCastInst>(BBI) && BBI->getType()->isPointerTy());
+      return BBI->isDebugOrPseudoInst();
```

Comment:
Inline the lambda, now that it's trivial?

Reply:
Wasn't sure if this should be updated to include assumes